File: Resources/CourseraAlgorithmicToolbox/week5_dynamic_programming1/4_longest_common_subsequence_of_two_sequences/lcs2.cpp
#include <iostream>
#include <vector>

using std::vector;

int lcs2(vector<int> &a, vector<int> &b) {
  //write your code here
  return std::min(std::min(a.size(), b.size()), c.size());
}

int main() {
  size_t n;
  std::cin >> n;
  vector<int> a(n);
  for (size_t i = 0; i < n; i++) {
    std::cin >> a[i];
  }

  size_t m;
  std::cin >> m;
  vector<int> b(m);
  for (size_t i = 0; i < m; i++) {
    std::cin >> b[i];
  }

  std::cout << lcs2(a, b) << std::endl;
}

File: Resources/CourseraAlgorithmicToolbox/week5_dynamic_programming1/4_longest_common_subsequence_of_two_sequences/LCS2.kt
import java.util.*
import kotlin.math.min

fun lcs2(a: IntArray, b: IntArray): Int {
    //Write your code here
    return min(a.size, b.size)
}

fun main(args: Array<String>) {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()
    val a = IntArray(n)
    for (i in 0 until n) {
        a[i] = scanner.nextInt()
    }

    val m = scanner.nextInt()
    val b = IntArray(m)
    for (i in 0 until m) {
        b[i] = scanner.nextInt()
    }

    println(lcs2(a, b))
}